### PR TITLE
types of routers has been modified

### PR DESCRIPTION
**Pull request by kravich13**

### CODE REVIEW

File: src/endpoints/crypto/coin-list.route.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+export const coinListRoute: RouteCustomOptions<{ Body: ICoinListBodyInput }> = {
+  url: '/coin-list',
+  method: 'POST',
+  schema: CoinListSchema,
+  handler: async (req, reply) => {
+    const { startDate, endDate, monthlyInvestment } = req.body;
+
+    const minDate = DateTime.fromISO('2013-01-01').toMillis();
```
Comment (mkbaranovskyi) [inline src/endpoints/crypto/coin-list.route.ts]: Magic string, move to configs

File: src/endpoints/crypto/coin-list.route.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    for (const { coinId, atl_date, image, symbol } of coins) {
+      const atlDate = DateTime.fromJSDate(atl_date).toMillis();
+
+      if (atlDate <= endDate) {
+        avialableCoins.push({ coinId, image, symbol });
+      }
+    }
+
+    const isExistsCryptoData = await CryptoDataEntity.findOneBy({ userId: String(user._id) });
```
Comment (mkbaranovskyi) [inline src/endpoints/crypto/coin-list.route.ts]: `cryptoData` or `cryptoDataInstance`

File: src/endpoints/crypto/coin-list.route.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    if (isExistsCryptoData) {
+      await CryptoDataEntity.update(user._id, {
```
Comment (mkbaranovskyi) [inline src/endpoints/crypto/coin-list.route.ts]: Use `save()` method to save the entity you already hold in your hands (from the line above)

File: src/endpoints/crypto/schemas/coin-list.schema.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  monthlyInvestment: number;
+}
+
+export const CoinListSchema = {
+  body: {
+    type: 'object',
+    properties: {
+      startDate: { type: 'number' },
+      endDate: { type: 'number' },
+      monthlyInvestment: { type: 'number' },
```
Comment (mkbaranovskyi) [inline src/endpoints/crypto/schemas/coin-list.schema.ts:13]: Some extra validation is needed: isPositive, isInteger

File: src/endpoints/crypto/schemas/coin-list.schema.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  startDate: number;
+  endDate: number;
+  monthlyInvestment: number;
+}
+
+export const CoinListSchema = {
+  body: {
+    type: 'object',
+    properties: {
+      startDate: { type: 'number' },
```
Comment (mkbaranovskyi) [inline src/endpoints/crypto/schemas/coin-list.schema.ts:11]: Some extra validation is needed: at least your 2013

File: src/endpoints/crypto/schemas/coin-list.schema.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    type: 'object',
+    properties: {
+      authorization: { type: 'string' },
+    },
+    required: ['authorization'],
+  },
+  response: {
+    200: {
+      type: 'array',
+      properties: {
```
Comment (mkbaranovskyi) [inline src/endpoints/crypto/schemas/coin-list.schema.ts]: Shouldn't is be an `items` property?

File: src/shared/database/entities/crypto-data.entity.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import { Column, Entity } from 'typeorm';
+import { Base } from './base.entity';
+
+@Entity('cryptoData')
+export class CryptoDataEntity extends Base {
+  @Column()
```
Comment (mkbaranovskyi) [inline src/shared/database/entities/crypto-data.entity.ts]: You don't have any relations between entities this way. You should use `@ManyToOne`, `@OneToOne`, etc.

**Review by mkbaranovskyi — CHANGES_REQUESTED** [review]

File: src/shared/database/entities/crypto-data.entity.ts (hunk 7 above)
Comment (kravich13) [reply src/shared/database/entities/crypto-data.entity.ts]: mongodb doesn't have any relations

**Review by mkbaranovskyi — APPROVED** [review]

**Pull request merged**